src/libs/actions/PaymentMethods.js

```diff
@@ -38,17 +38,35 @@ function getPaymentMethods() {
  * @param {Object} params
  */
 function addBillingCard(params) {
+    const cardYear = params.expirationDate.substr(3);
+    const cardMonth = params.expirationDate.substr(0, 2);
+
     API.AddBillingCard({
         cardNumber: params.cardNumber,
-        cardYear: params.expirationDate.substr(3),
-        cardMonth: params.expirationDate.substr(0, 2),
+        cardYear,
+        cardMonth,
         cardCVV: params.securityCode,
         addressName: params.nameOnCard,
         addressZip: params.zipCode,
         currency: CONST.CURRENCY.USD,
     }).then(((response) => {
         if (response.jsonCode === 200) {
-            Onyx.set(ONYXKEYS.CARD_LIST, response);
+            const cardObject = {
+                additionalData: {
+                    isBillingCard: true,
+                    isP2PDebitCard: false,
+                },
+                addressName: params.nameOnCard,
+                addressState: params.selectedState,
+                addressStreet: params.billingAddress,
+                addressZip: params.zipCode,
+                cardMonth,
+                cardNumber: params.cardNumber,
```

Comment:
We need to mask all but the first 4 and last 4 digits here. There might be a util somewhere I'm not sure

Reply:
Do you mean we want to save the card in ONYX in the format 4242 **** **** 4242?

Reply:
Yes but with x's like the example above. `"520082XXXXXX8210"`